name/views.py

```diff
@@ -145,463 +73,108 @@ def item_location(self, item):
         return location_set
 
     def item_description(self, item):
-        return "Name Type: %s" % NAME_TYPE_CHOICES[item.name_type][1]
+        return "Name Type: %s" % item.get_name_type_label()
 
     def item_link(self, item):
         return VOCAB_DOMAIN + 'name/' + item.name_id + '/'
 
 
 def entry_detail(request, name_id):
-    """
-    Renders the requested user detailed info page
-    """
+    """View for the Name detail page."""
+    queryset = (
+        Name.objects.select_related().
+        prefetch_related('identifier_set__type')
+    )
 
-    # define the name_entry from the passed id
-    name_entry = get_object_or_404(Name, name_id=name_id)
+    name_entry = get_object_or_404(queryset, name_id=name_id)
 
-    # If the user is merged with any other user, set the
-    # name_entry to the merged user and redirect.
     if name_entry.merged_with:
-        name_entry = (
-            get_object_or_404(Name, name_id=name_entry.merged_with))
-
-        return HttpResponseRedirect(
-            reverse('name_entry_detail', args=[name_entry.name_id]))
-
-    # if suppressed record, return 'not found' code 404
-    elif name_entry.record_status == 2:
-        return HttpResponse(status=404)
-
-    # if deleted record, return 'gone' code 410
-    elif name_entry.record_status == 1:
-        return HttpResponseGone('The requested record has been deleted!')
-
-    else:
-        if name_entry.name_type == 4:
-            try:
-                locations = Location.objects.filter(
-                    belong_to_name=name_entry)
-                current_location = Location.objects.get(
-                    belong_to_name=name_entry, status=0)
-            except:
-                locations = None
-                current_location = None
-        else:
-            current_location = None
-            locations = None
-
-        date_display = DATE_DISPLAY_LABELS[name_entry.name_type]
-        total_entries = Name.objects.all()
-        ordered_link_set = name_entry.identifier_set.order_by('order')
-        note_set = name_entry.note_set.exclude(note_type=2)
-
-        return render_to_response(
-            'name/name_detail.html',
-            {
-                'total_entries': total_entries,
-                'requested_user': name_entry,
-                'current_location': current_location,
-                'locations': locations,
-                'note_set': note_set,
-                'ordered_link_set': ordered_link_set,
-                'types': dict(NAME_TYPE_CHOICES),
-                'date_display_begin': date_display['begin'],
-                'date_display_end': date_display['end'],
-            },
-            context_instance=RequestContext(request)
-        )
+        return redirect(name_entry.merged_with)
+
+    elif name_entry.is_suppressed():
+        return http.HttpResponseNotFound()
```

Comment:
Perhaps, instead of sending back a 410 or a 404 depending on the name record status, we just return a 404 if the name is not active. I feel like @vphill, will have more insight on this.